plugins/connection/incus.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+# Based on lxd.py (c) 2016, Matt Clay <matt@mystile.com>
+# (c) 2023, Stephane Graber <stgraber@stgraber.org>
+# Copyright (c) 2023 Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Stéphane Graber <stgraber@stgraber.org>
+    connection: incus
+    short_description: Run tasks in Incus instances via the Incus CLI
+    description:
+        - Run commands or put/fetch files to an existing Incus instance using Incus CLI
+    version_added: "2.0"
+    options:
+      remote_addr:
+        description:
+            - Instance identifier
+        default: inventory_hostname
+        vars:
+            - name: ansible_host
+            - name: ansible_incus_host
+      executable:
+        description:
+            - shell to use for execution inside the instance
+        default: /bin/sh
+        vars:
+            - name: ansible_executable
+            - name: ansible_incus_executable
+      remote:
+        description:
+            - name of the Incus remote to use
+        default: local
+        vars:
+            - name: ansible_incus_remote
+      project:
+        description:
+            - name of the Incus project to use
+        default: default
+        vars:
+            - name: ansible_incus_project
+"""
+
+import os
+from distutils.spawn import find_executable
+from subprocess import call, Popen, PIPE
+
+from ansible.errors import AnsibleError, AnsibleConnectionFailure, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_text
+from ansible.plugins.connection import ConnectionBase
+
+
+class Connection(ConnectionBase):
+    """ Incus based connections """
+
+    transport = "incus"
+    has_pipelining = True
+    default_user = 'root'
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
+
+        self._host = self._play_context.remote_addr.split(".")[0]
+        self._incus_cmd = find_executable("incus")
+
+        if not self._incus_cmd:
+            raise AnsibleError("incus command not found in PATH")
+
+        if self._play_context.remote_user is not None and self._play_context.remote_user != 'root':
```

Comment:
In general `self._play_context` should be avoided if possible. Since `remote_user` is not an option for this plugin, I would simply remove this check.